### PR TITLE
Lock SDK to 6.0.100 and ignore with Renovate updates

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

Disable renovate from bumping the dotnet-sdk. We have prefer latest which should be sufficient.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)

### CODE REVIEW

**Comment by withinfocus** [comment]
Is https://github.com/bitwarden/sdk/pull/286 an example? I don't see the harm really despite the `rollForward`.

**Comment by Hinton** [comment]
@withinfocus yes. It's an annoyance since the dotnet tooling will complain about missing SDK version. Requiring developers to always be on the latest version can be somewhat tedious.

**Comment by withinfocus** [comment]
You mean the runners not being able to pull them? On a weekly cadence that seems rare since these get out in a day or two.

Your point is certainly valid since this will still roll up to the latest release, but _when_ that's available could actually be of importance to us and our team / processes. Microsoft may issue an SDK update here for security reasons and we want to force the updates and conversation if we can; the potentially week-long delay for the update is already large and we want to push upgrades wherever as possible as quickly as possible. If we weren't in the security software space I think this change would be an easier sell, but I am taking pause here because in our seat we want to push harder.

**Comment by Hinton** [comment]
No, I'm referring to the developer workflow. A developer should not have to update their SDK every time a patch released is issued. We should enforce that the deployed code is built with appropriate releases though. But that's outside the scope of the `global.json`.

**Comment by withinfocus** [comment]
Hmm not sure I agree -- we want our developers updated too, and this is ease to automate. With one regular command that I run regularly locally I get these updates (https://github.com/isen-ng/homebrew-dotnet-sdk-versions).

**Comment by Hinton** [comment]
I generally try and avoid taps and installs it directly from Microsoft to ensure the integrity.

**Comment by withinfocus** [comment]
Relates to https://github.com/bitwarden/server/pull/3478 -- we are opting for an easier developer experience as of now.

File: .github/renovate.json [hunk 1]
```diff
@@ -45,5 +45,6 @@
       "matchManagers": ["github-actions"],
       "matchUpdateTypes": ["minor", "patch"]
     }
-  ]
+  ],
```
Comment (withinfocus) [inline .github/renovate.json:48]: ℹ️ If lockfiles show up for the .NET SDK here (I hope they don't) then we may need to force a Renovate constraint.

File: languages/csharp/global.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "6.0.413",
+    "version": "6.0.100",
```
Comment (withinfocus) [inline languages/csharp/global.json:3]: ℹ️ I may take a quick look at an SDK upgrade to 8, unless you foresee an issue.

**Review by withinfocus — APPROVED** [review]

File: languages/csharp/global.json (hunk 2 above)
Comment (Hinton) [reply languages/csharp/global.json:3]: As long as the package remains compatible with all actively supported .NET installs. We may need to explore adding .net standard support in the future.

**Pull request merged**